File: buggy3.ts
function calculateTotal(items: number[]): number {
    let total = 0;

    for (let i = 0; i <= items.length; i++) {
        total += items[i];
    }

    return total;
}

function getUser(users: { name: string; age: number }[], name: string) {
    return users.find(user => user.name === name);
}

const users = [
    { name: "Jayant", age: 20 },
    { name: "Alex", age: 21 }
];

console.log(calculateTotal([10, 20, 30]));
console.log(getUser(users, "John").age);

File: buggy3.py
def calculate_average(numbers):
    total = 0

    for n in numbers:
        total += n

    return total / len(numbers)


def find_user(users, name):
    for user in users:
        if user["name"] == name:
            return user

    return None


users = [
    {"name": "Jayant", "age": 20},
    {"name": "Alex", "age": 21},
]

print(calculate_average([]))
print(find_user(users, "John")["age"])